ncm-syslog/src/main/perl/syslog.pm

```diff
@@ -1,253 +1,243 @@
-		      }
-		  }
-		  else {
-		      $self->debug(2,"facility $facility is not yet using action $action");
-		      # this facility has not yet used this action, simply add it to this action
-		      $changes += map { s/\s*${action}/;${facility}\.${priority}\t${action}/ unless /^#/ } @syslogcontents;
-		  }
-	      }
-	      else{
-		  $self->debug(2,"action $action is not yet known");
-		  # this action not known, just add
-		  $changes ++;
-		  push @syslogcontents, "$facility.$priority\t$action\n";
-	      }
-	  }
-	  # next selector
-	  $selectors ++;
-	  $selectorpath = "$configpath/selector/$selectors";
-      }
-      if ( $fullcontrol ){
-	  push @syslogcontents, "$line\t$action\n";
-      }
-      # next entry;
-      $entry ++;
-      $configpath     = ${basepath}."/config/$entry";
-  }
-  if ( $fullcontrol ){
-      open SLC, "+>$syslogconf.new";
-      print SLC @directives;
-      print SLC @syslogcontents;
-      close SLC;
-      $changes = system ("/usr/bin/diff", "-q", "$syslogconf.new", "$syslogconf"); 
-      if ( $changes ){
-          $self->debug(2,"there are changes to be made to $syslogconf");
-          my $rc = system("/bin/mv", "$syslogconf.new", "$syslogconf");
-      }
-  }
-  elsif ($changes){
-      open SLC, "+>$syslogconf";
-      print SLC @directives;
-      print SLC @syslogcontents;
-      close SLC;
-  };
-
-# now the options for the daemons
-  $syslogconf = "/etc/sysconfig/$daemontype";
-  $basepath = "/software/components/syslog/syslogdoptions";
-  if ($config->elementExists($basepath)) {
-    my $options = $config->getValue($basepath);
-    $changes+=NCM::Check::lines($syslogconf,
-                  linere => "^\\s*SYSLOGD_OPTIONS.*",
-                  goodre => "^\\s*SYSLOGD_OPTIONS\\s*=\\s*\"$options\"",
-                  good   => "SYSLOGD_OPTIONS=\"$options\"" );
-  }
-  $basepath = "/software/components/syslog/klogdoptions";
-  if ($config->elementExists($basepath)) {
-    my $options = $config->getValue($basepath);
-    $changes+=NCM::Check::lines($syslogconf,
-                  linere => "^\\s*KLOGD_OPTIONS.*",
-                  goodre => "^\\s*KLOGD_OPTIONS\\s*=\\s*\"$options\"",
-                  good   => "KLOGD_OPTIONS=\"$options\"" );
-  }
-
-  if ($changes){
-    system("/sbin/service $daemontype restart");
-  }
-
-  return;
-}
-##########################################################################
-sub Unconfigure {
-##########################################################################
-  my ($self,$config)=@_;
-  $self->info("Unconfigure does nothing for $0\n");
-  return;
+#${PMpre} NCM::Component::syslog${PMpost}
+
+=head1 NAME
+
+C<NCM::Component::syslog> configures entries in /etc/(r)syslog.conf
+
+=head1 Methods
+
+=over
+
+=cut
+
+use parent qw(NCM::Component);
+
+use CAF::FileWriter;
+use CAF::FileEditor 16.12.1;
+use CAF::Service;
+use Readonly;
+
+Readonly my $SYSLOG_DEFAULT_DAEMONTYPE => 'syslog';
```

Comment:
i'll also default this in pan; but i have to keep this in the code as well, as we don't enforce schema and component versions.